engine/paima-providers/src/evm.ts

```diff
@@ -199,7 +202,8 @@ export class EvmProvider implements IProvider<EvmApi> {
       }
       return hash;
     } catch (e: any) {
-      throw new ProviderApiError(`[switchChain] error: ${e?.message}`);
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      throw new ProviderApiError(`[switchChain] error: ${e?.message}`, e?.code);
```

Comment:
I noticed I made a copy-paste mistake here too. The error should be `sendTransaction` and not `switchChain`